aws/resource_aws_acm_certificate_test.go

```diff
@@ -263,10 +263,10 @@ func TestAccAWSAcmCertificate_root_TrailingPeriod(t *testing.T) {
 				Config: testAccAcmCertificateConfig(domain, acm.ValidationMethodDns),
 				Check: resource.ComposeTestCheckFunc(
 					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "acm", regexp.MustCompile(`certificate/.+`)),
-					resource.TestCheckResourceAttr(resourceName, "domain_name", strings.TrimSuffix(domain, ".")),
+					resource.TestCheckResourceAttr(resourceName, "domain_name", trimTrailingPeriod(domain)),
```

Comment:
The `TestAccAWSAcmCertificate_root_TrailingPeriod` testing can now be `ExpectError` to match the API's expectations. 👍